src/Common/Scheduler/ISchedulerNode.h

```diff
@@ -79,6 +81,16 @@ struct SchedulerNodeInfo
         weight = value;
     }
 
+    void setQueueSize(Int64 value)
+    {
+        if (value <= 0)
+            throw Exception(
+                ErrorCodes::INVALID_SCHEDULER_NODE,
+                "Zero, negative and non-finite node queue_size are not allowed: {}",
```

Comment:
Let's make error message more user-friendly:
```suggestion
                "Workload setting `max_waiting_queries` value must be positive, got: {}",
```